src/components/ColorPicker/ColorPicker.tsx

```diff
@@ -0,0 +1,470 @@
+          variant === 'circle' && classes.colorCircle,
+          variant === 'square' && classes.colorSquare
+        )}
+        style={{
+          backgroundColor: isValidHexColor(color) ? color : '#000000',
+        }}
+      />
+    );
+
+    const handleChange = (color: string) => {
+      setColorValue(color);
+      onChange?.(color);
+    };
+
+    return (
+      <div className={clsx(classes.root, className)}>
+        {label && (
+          <label
+            className={clsx(
+              classes.label,
+              color === 'inverse' && classes.labelInverse
+            )}
+            htmlFor={uniqueId}
+          >
+            {label}
+            {!!Icon && tooltipMessage && (
+              <Tooltip title={tooltipMessage}>
+                <span className={classes.tooltipContainer}>
+                  <Icon
+                    aria-hidden
+                    className={clsx(
+                      classes.labelIcon,
+                      color === 'inverse' && classes.labelIconInverse
+                    )}
+                    height={16}
+                    width={16}
+                    role="img"
+                  />
+                </span>
+              </Tooltip>
+            )}
+            {secondaryLabel && (
+              <span
+                className={clsx(
+                  classes.labelSecondary,
+                  color === 'inverse' && classes.labelInverse
+                )}
+              >
+                {secondaryLabel}
+              </span>
+            )}
+          </label>
+        )}
+        <div
+          className={clsx(
+            classes.inputContainer,
+            fullWidth && classes.inputFullWidth
+          )}
+        >
+          <input
+            aria-describedby={buildDescribedBy({
+              hasError,
+              hasHelpMessage: !!helpMessage,
+              uniqueId,
+            })}
+            aria-label={ariaLabel}
+            className={clsx(
+              classes.input,
+              fullWidth && classes.inputFullWidth,
+              hasError && classes.inputError,
+              hasError && color === 'inverse' && classes.inputErrorInverse,
+              { [classes.inputInverse]: color === 'inverse' }
+            )}
+            disabled={disabled}
+            id={uniqueId}
+            name={name}
+            onChange={(e) => handleChange(e.target.value)}
+            readOnly={readOnly}
+            ref={ref}
+            type="text"
+            value={colorValue}
+            {...rootProps}
+          />
+
+          <Popover
+            anchorElement={
+              <ButtonUnstyled
+                aria-label="Pick color"
+                disabled={disabled || readOnly}
+              >
+                <Color className={classes.colorPosition} color={colorValue} />
+              </ButtonUnstyled>
+            }
+            aria-label="Color Picker"
+          >
+            {({ popover }: { popover: any }) => (
```

Comment:
I could be wrong, but I think we export a type for the render props here:

```suggestion
            {({ popover }: PopoverRenderProps) => (
```

Reply:
Ooo for some reason I only tried `PopoverReturnProps`. Thank you! 🙏